docs/advanced.rst

```diff
@@ -1589,24 +1589,56 @@ It is also possible to call python functions via ``operator()``.
     py::object result_py = f(1234, "hello", some_instance);
```

Comment:
What do you think about creating a new page dedicated to the Python C++ API? The wrapper classes (`str`, `tuple`, `dict`) are also a bit underdocumented, but the the Advanced page is already getting quite long and hard to navigate. (It's probably worth splitting of additional sections which form a logical unit (like buffer/numpy/eigen) but that's a topic for a different PR.)